Framework/Algorithms/src/ReflectometryReductionOne.cpp

```diff
@@ -278,6 +278,50 @@ void ReflectometryReductionOne::init() {
                   Direction::Input);
 }
 
+//----------------------------------------------------------------------------------------------
+/** Validate the input properties.
+*/
+std::map<std::string, std::string> ReflectometryReductionOne::validateInputs() {
+
+  std::map<std::string, std::string> result;
+
+  MatrixWorkspace_sptr decWS = getProperty("DetectorEfficiencyCorrection");
+  MatrixWorkspace_sptr runWS = getProperty("InputWorkspace");
+
+  if (!decWS) {
+    return result;
+  }
+
+  if (decWS->getAxis(0)->unit()->unitID() != "Wavelength") {
+    result["DetectorEfficiencyCorrection"] = "Detector Efficiency Correction "
+      "workspace x-units must be in wavelength";
+  }
+  else if (decWS->getNumberHistograms() > 1 &&
+      decWS->getNumberHistograms() < runWS->getNumberHistograms()) {
+    result["DetectorEfficiencyCorrection"] = "The number of spectra in the "
+      "Detector Efficiency Correction workspace must either be 1 or equal to "
+      "the number of spectra in the input workspace";
```

Comment:
If the number of spectra must be either 1 or equal to the number of spectra in the input workspace, shouldn't we be checking:

```
else if (decWS->getNumberHistograms() > 1 
&&
decWS->getNumberHistograms() != runWS->getNumberHistograms())
```